### PR TITLE
Add Role, Permission and Admin Entities

### PR DESCRIPTION
**Pull request by Gabriel-Rockson**
- chore: Add configuration ofor Spring data jpa and Flyway
- chore: Add docker compose file to setup database
- feat: Add Role and Permission entities
- feat: Add Many to Many relationship between Roles and Permissions
- chore: Add dependencies for validation
- feat: Add Admin entity and relationship between Role entity and Admin
- chore: Remove db secrets from application.yml

### CODE REVIEW

File: src/main/java/com/amalitech/mentorshiptrackr/models/Admin.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.UUID;
+
+@Entity
+@Table(name = "admins")
+public class Admin {
```
Comment (daniel-bontii) [inline src/main/java/com/amalitech/mentorshiptrackr/models/Admin.java:12]: @Getter and @Setter can be applied at the class level. This way you don't have to add it to all the fields

Comment (Gabriel-Rockson) [reply src/main/java/com/amalitech/mentorshiptrackr/models/Admin.java:18]: Made changes

File: src/main/java/com/amalitech/mentorshiptrackr/models/Permission.java [hunk 2]
```diff
@@ -19,15 +20,12 @@ public class Permission {
     @Getter
```
Comment (daniel-bontii) [inline src/main/java/com/amalitech/mentorshiptrackr/models/Permission.java:20]: Remove this duplicate getter and then you can merge your work afterwards

Comment (Gabriel-Rockson) [reply src/main/java/com/amalitech/mentorshiptrackr/models/Permission.java:20]: Alright

**Pull request merged**